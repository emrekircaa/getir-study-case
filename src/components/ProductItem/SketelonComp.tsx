import React from "react";
import ContentLoader, { IContentLoaderProps } from "react-content-loader";

const Skeleton: React.FC<IContentLoaderProps> = (props) => (
  <ContentLoader
    speed={2}
    width={124}
    height={225}
    viewBox="0 0 124 225"
    backgroundColor="#f3f3f3"
    foregroundColor="#ecebeb"
    {...props}
  >
    <rect x="-3" y="2" rx="12" ry="12" width="124" height="124" />
    <rect x="127" y="48" rx="3" ry="3" width="53" height="11" />
    <rect x="187" y="48" rx="3" ry="3" width="72" height="11" />
    <rect x="2" y="190" rx="3" ry="3" width="112" height="19" />
    <rect x="265" y="191" rx="3" ry="3" width="37" height="11" />
    <rect x="166" y="23" rx="3" ry="3" width="173" height="11" />
    <rect x="1" y="162" rx="3" ry="3" width="100" height="21" />
    <rect x="1" y="137" rx="3" ry="3" width="70" height="15" />
  </ContentLoader>
);

export default Skeleton;
