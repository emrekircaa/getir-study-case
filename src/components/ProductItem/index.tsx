import React, { memo } from "react";
import styled from "styled-components";
import { colors } from "../../theme";
import { Button } from "../Button";
import defaultImage from "../../assets/image/defaultImage.png";
interface ProductItem {
  price: number;
  name: string;
  onAddButtonClick: () => void;
}

export const ProductItem: React.FC<ProductItem> = memo(
  ({ price, name, onAddButtonClick }) => {
    return (
      <StyledProductItemCard data-testid="ProductItem">
        <StyledImageWrapper>
          <img src={defaultImage} alt={name} />
        </StyledImageWrapper>
        <StyledProductItemTextContainer>
          <StyledPriceText>
            <span>â‚º</span> {price.toFixed(2)}
          </StyledPriceText>
          <StyledNameText>{name}</StyledNameText>
        </StyledProductItemTextContainer>

        <Button onClick={() => onAddButtonClick()}>Add</Button>
      </StyledProductItemCard>
    );
  }
);

const StyledProductItemTextContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

const StyledProductItemCard = styled.div`
  width: 124px;
  display: flex;
  flex-direction: column;
  gap: 8px;
  @media (max-width: 425px) {
    width: 46%;
  }
  @media (max-width: 360px) {
    width: 100%;
  }
`;
const StyledImageWrapper = styled.div`
  height: 124px;
  width: 124px;
  aspect-ratio: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 16px;
  border-radius: 12px;
  background-color: #fefefe;
  border: 1.18px solid #f3f0fe;
  img {
    object-fit: cover;
    width: 100%;
    height: 100%;
  }
`;
const StyledPriceText = styled.div`
  font-weight: 700;
  font-size: 14px;
  height: 23px;
  color: ${colors.primary};
  text-align: left;

  span {
    font-weight: 400;
  }
`;
const StyledNameText = styled.div`
  font-weight: 600;
  display: -webkit-box;
  width: 100%;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
  height: 40px;
  font-size: 14px;
  line-height: 20px;
  text-align: left;
  color: ${colors.gray700};
`;
