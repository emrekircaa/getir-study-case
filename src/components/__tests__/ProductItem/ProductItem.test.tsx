import { render, fireEvent } from "@testing-library/react";
import { ProductItem } from "../../ProductItem";
import React from "react";

describe("ProductItem component", () => {
  it("renders with correct props", () => {
    const onAddButtonClick = jest.fn();
    const { getByTestId } = render(
      <ProductItem
        price={10.99}
        name="Product A"
        onAddButtonClick={onAddButtonClick}
      />
    );
    const productItem = getByTestId("ProductItem");
    expect(productItem).toMatchSnapshot();
  });

  it("calls onAddButtonClick when Add button is clicked", () => {
    const onAddButtonClick = jest.fn();
    const { getByRole } = render(
      <ProductItem
        price={10.99}
        name="Product A"
        onAddButtonClick={onAddButtonClick}
      />
    );

    const addBtn = getByRole("button");
    fireEvent.click(addBtn);
    expect(onAddButtonClick).toBeCalledTimes(1);
  });
});
