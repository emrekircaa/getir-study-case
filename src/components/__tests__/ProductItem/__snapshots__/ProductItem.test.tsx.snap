// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem component renders with correct props 1`] = `
<div
  class="sc-dmyCSP fdHOWZ"
  data-testid="ProductItem"
>
  <div
    class="sc-hLQSwg jJSUZT"
  >
    <img
      alt="Product A"
      src="defaultImage.png"
    />
  </div>
  <div
    class="sc-guDLey dDYrwN"
  >
    <div
      class="sc-eDLKkx gcxRue"
    >
      <span>
        â‚º
      </span>
       
      10.99
    </div>
    <div
      class="sc-jTQCzO hopybM"
    >
      Product A
    </div>
  </div>
  <button
    class="sc-beySPh hqIXqv"
  >
    Add
  </button>
</div>
`;
